strawberryfields/apps/similarity.py

```diff
@@ -306,33 +320,148 @@ def event_cardinality(photon_number: int, max_count_per_mode: int, modes: int) -
     return cardinality
 
 
-def prob_orbit_mc(
-    graph: nx.Graph, orbit: list, n_mean: float = 5, samples: int = 1000, loss: float = 0.0
+def _prob_orbit_exact(
+    graph: nx.Graph,
+    orbit: list,
+    n_mean: float = 5,
+    loss: float = 0.0
 ) -> float:
-    r"""Gives a Monte Carlo estimate of the GBS probability of a given orbit according to the input
+    r"""Gives the exact GBS probability of a given orbit according to the input
     graph.
 
-    To make this estimate, several samples from the orbit are drawn uniformly at random using
-    :func:`orbit_to_sample`. The GBS probabilities of these samples are then calculated and the
-    sum is used to create an estimate of the orbit probability.
+    The exact probability of an orbit is made up of the sum of probabilities of
+    all possible GBS output patterns that belong to it:
+
+    .. math::
+       P(O) = \sum_{n \in O} P(n) = \frac{{\rm haf}(A_{n})^2}{n! \sqrt{\det(Q)}}
+
+    where :math:`n` represents one GBS output pattern, :math:`A_{n}` is its induced
+    sub-matrix and :math:`Q` is the Q-matrix obtained from the adjacency matrix.
+
+    **Example usage:**
+
+    >>> graph = nx.complete_graph(8)
+    >>> _prob_orbit_exact(graph, [2, 1, 1])
+    0.03744

```

Comment:
Note that since this is a private function we probably don't need to do example code blocks.
But I take the view that it can only help, so :+1: from me!

Reply:
I had the same thought :) It might help if someone looks into the source code.